fitstack/signal.py

```diff
@@ -957,9 +970,19 @@ def __init__(
         derivs: Optional[Dict[str, Tuple[float, float]]] = None,
         convolutions: Optional[Dict[str, Tuple[float, float]]] = None,
         kpar_range: Optional[Tuple[float, float]] = None,
+        nu21: float = 1420.40575177,  # MHz
+        z_eff: Optional[float] = None,  # effective redshift
+        cs: float = 299792.458,  # speed of light in km/s          
         *args,
         **kwargs,
     ):
+        
+        # Set default z_eff if None
+        self.z_eff = z_eff if z_eff is not None else 1.0
+        cosmo_cora = cosmology.Cosmology()
+        self.H_z = cosmo_cora.H(self.z_eff) * u.mega_parsec / 1000.  # In km/s/Mpc
+        self.nu21 = nu21 
+        self.cs = cs           
```

Comment:
```suggestion
        
        # Set default z_eff if None
        self.z_eff = z_eff if z_eff is not None else 1.0
        cosmo_cora = cosmology.Cosmology()
        self.H_z = cosmo_cora.H(self.z_eff) * u.mega_parsec / 1000.  # In km/s/Mpc
```
Remove local values of `nu21` and `cs`